packages/react-vapor/src/components/select/hoc/tests/MultiSelectWithPredicate.spec.tsx

```diff
@@ -41,168 +26,81 @@ describe('Select', () => {
-            });
-        });
-
-        it('should hide items when they do not match the predicates', () => {
-            const items = [{value: 'a'}, {value: 'b', selected: true}, {value: 'c', selected: true}];
-
-            mountMultiSelect({items});
-            store.dispatch(selectFlatSelect(id, defaultFlatSelectOptions[1].id));
-            wrapper.update();
-            multiSelect = wrapper.find(SelectConnected);
-
-            expect(multiSelect.props().items.length).toBe(items.length);
-            multiSelect
-                .find(SelectConnected)
-                .props()
-                .items.every((item: IItemBoxProps) => {
-                    expect(item.hidden).toBe(true);
-                });
+            lists = screen.getAllByRole('list');
+            expect(within(lists[1]).queryByText('first')).not.toBeInTheDocument();
+            expect(within(lists[1]).getByText('No Items')).toBeInTheDocument();
         });
 
-        it('should not show items that are already hidden', () => {
-            jest.spyOn(FlatSelectSelectors, 'getSelectedOptionId').mockReturnValue(defaultFlatSelectOptions[0].id);
-
-            const items = [
-                {value: 'a', hidden: true},
-                {value: 'b', selected: true},
-                {value: 'c', selected: true},
+        it('shows items that match the predicates', () => {
+            const options: IFlatSelectOptionProps[] = [
+                {id: 'my_real_id_01', option: {content: 'All'}, selected: true},
+                {id: 'my_real_id_02', option: {content: 'Beer'}},
             ];
+            const matcher = (predicate: string, item: IItemBoxProps) =>
+                predicate === defaultFlatSelectOptions[0].id || item.value === '🍻';
+            const items = [{value: '🐝'}, {value: '🍻'}, {value: '🥩'}];
 
-            const multiSelectWrapper = mountWithStore(
-                <MultiSelectWithPredicate {...basicProps} items={items} />,
-                getStoreMock({})
-            );
+            render(<MultiSelectWithPredicate id={id} options={options} items={items} matchPredicate={matcher} />);
+            // open the dropdown
+            userEvent.click(screen.getByRole('button', {name: 'Select an option'}));
 
-            multiSelect = multiSelectWrapper.find(SelectConnected);
+            // select the all flat select
+            userEvent.click(screen.getByText('Beer'));
 
-            expect(multiSelect.props().items.length).toBe(items.length);
-            expect(multiSelect.find(SelectConnected).props().items[0].hidden).toBe(true);
-            expect(multiSelect.find(SelectConnected).props().items[1].hidden).toBeUndefined();
-            expect(multiSelect.find(SelectConnected).props().items[2].hidden).toBeUndefined();
+            const lists = screen.getAllByRole('list');
+            expect(within(lists[1]).queryByText('🐝')).not.toBeInTheDocument();
+            expect(within(lists[1]).getByText('🍻')).toBeInTheDocument();

```

Comment:
Have you tried `.toBeVisible`? I know it takes into account other stuff like hidden attributes and even css (if loaded)

Reply:
👍 I've switch them all, it's easier to read